From fde76731a6589bffbc56ddbd31042c49b2e45003 Mon Sep 17 00:00:00 2001
From: Karthik M <quic_karm@quicinc.com>
Date: Fri, 12 Jan 2024 22:12:34 +0530
Subject: [PATCH] hostapd: Reset the EML capability advertisement for AP

As per IEEE802.11 be draft 5.0, eml transistion delay and padding delay
capability sent by FW should be resetted to zero for AP MLD.

These capabilities are non AP MLD specific configurations

Change-Id: I4d3202eeaae371d6c870c9d48ef9022c8c6e895f
Signed-off-by: Karthik M <quic_karm@quicinc.com>
---
--- a/src/ap/ieee802_11_eht.c
+++ b/src/ap/ieee802_11_eht.c
@@ -616,6 +616,12 @@ u8 * hostapd_eid_eht_basic_ml(struct hos
 	wpabuf_put_u8(buf, hapd->rx_cu_param.bpcc);
 
 	if (!hapd->conf->disable_eml_cap) {
+		/* Reset the EMLSR Transision and Padding delay to zero for
+		 * MLD AP as per IEEE802.11 be draft 5.0
+		 */
+		hapd->iface->mld_eml_capa &= ~EHT_ML_EML_CAPA_EMLSR_TRANS_DELAY_MASK;
+		hapd->iface->mld_eml_capa &= ~EHT_ML_EML_CAPA_EMLSR_PADDING_DELAY_MASK;
+
 		wpa_printf(MSG_DEBUG, "MLD: EML Capabilities=0x%x",
 			   hapd->iface->mld_eml_capa);
 		wpabuf_put_le16(buf, hapd->iface->mld_eml_capa);
-- 
2.17.1

